File: data_structure/KimJunYoung/ProgrammersStackQueue1.java
package data_structure.KimJunYoung;

import java.util.*;
public class ProgrammersStackQueue1 {
    public static void main(String[] args) {
        ProgrammersStackQueue1 solved = new ProgrammersStackQueue1();
        int[] testCase1 = {9, 9, 9};
        System.out.println(Arrays.toString(solved.solution(testCase1))); // [9]
        int[] testCase2 = {1};
        System.out.println(Arrays.toString(solved.solution(testCase2))); // [1]
        int[] testCase3 = {1, 2, 1, 2, 1, 2, 2, 1, 2, 1, 1};
        System.out.println(Arrays.toString(solved.solution(testCase3))); // [1, 2, 1, 2, 1, 2, 1, 2, 1]
    }
    public int[] solution(int []arr) {
        List<Integer> numbers = new ArrayList<>();
        numbers.add(arr[0]);
        for (int num : arr) {
            int before = numbers.get(numbers.size() - 1);
            if (num != before) {
                numbers.add(num);
            }
        }

        return numbers.stream().mapToInt(x -> x).toArray();
    }
}

File: data_structure/dowon/programmers_stack_queue_1.py
def solution(arr):
    answer = []
    answer.append(arr[0])  # O(1)
    for i in arr[1:]:      # O(n)
        if answer[-1] != i:  # O(1) answer의 마지막 요소와 현재 요소를 비교
            answer.append(i)  # O(1): 중복되지 않는 경우 현재 요소를 answer에 추가
    return answer  # O(1)